include/tudat/simulation/estimation_setup/observationOutputSettings.h

```diff
@@ -32,22 +34,117 @@ enum ObservationDependentVariables
-        variableType_( variableType )
+            const ObservationDependentVariables variableType,
+            const LinkEndId linkEndId = LinkEndId( "", "" ),
+            const LinkEndType linkEndType = unidentified_link_end,
+            const LinkEndId originatingLinkEndId = LinkEndId( "", "" ),
+            const LinkEndType originatingLinkEndType = unidentified_link_end ):
+        variableType_( variableType ), linkEndId_( linkEndId ), linkEndType_( linkEndType ),
+        originatingLinkEndId_( originatingLinkEndId ), originatingLinkEndType_( originatingLinkEndType ){ }
+
+    virtual ~ObservationDependentVariableSettings( ){ }
+
+    ObservationDependentVariables variableType_;
+
+    std::string getBaseIdentifier( )
     {
+        std::string identifier = ", link end";
+        if( linkEndId_ != LinkEndId( "", "" ) && linkEndType_ != unidentified_link_end )
+        {
+            identifier += ": (" + linkEndId_.bodyName_ + ", " + linkEndId_.stationName_ + ") as " + getLinkEndTypeString( linkEndType_ );
+        }
+        else if ( linkEndId_ != LinkEndId( "", "" ) )
+        {
+            identifier = ": (" + linkEndId_.bodyName_ + ", " + linkEndId_.stationName_ + ")";
+        }
+        else if ( linkEndType_ != unidentified_link_end )
+        {
+            identifier = " of type " + getLinkEndTypeString( linkEndType_ );
+        }
 
+        if( originatingLinkEndId_ != LinkEndId( "", "" ) && originatingLinkEndType_ != unidentified_link_end )
+        {
+            identifier += " to link end: (" + originatingLinkEndId_.bodyName_ + ", " + originatingLinkEndId_.stationName_ + ") as " + getLinkEndTypeString( originatingLinkEndType_ );
+        }
+        else if ( originatingLinkEndId_ != LinkEndId( "", "" ) )
+        {
+            identifier += " to link end: (" + originatingLinkEndId_.bodyName_ + ", " + originatingLinkEndId_.stationName_ + ")";
+        }
+        else if ( originatingLinkEndType_ != unidentified_link_end )
+        {
+            identifier += " to link end of type " + getLinkEndTypeString( originatingLinkEndType_ );
+        }
+        return identifier;
     }
 
-    virtual ~ObservationDependentVariableSettings( ){ }
+    virtual std::string getIdentifier( )
+    {
+        return getBaseIdentifier( );
+    }
 
-    ObservationDependentVariables variableType_;
+    virtual bool areSettingsCompatible( const std::shared_ptr< ObservationDependentVariableSettings > otherSettings )
+    {
+        return areBaseSettingsCompatible( otherSettings, true );
+    }
+
+    bool areBaseSettingsCompatible( const std::shared_ptr< ObservationDependentVariableSettings > otherSettings,
+                                    const bool revertedLinksAllowed = true )
+    {
+        bool isCompatible = true;
+        if ( variableType_ != otherSettings->variableType_ )
+        {
+            isCompatible = false;
+        }
+        else
+        {
+            bool directLinksMatch =
```

Comment:
could you add some comments to what this does or (even better) split it up into separate bools to make it explicit what is being checked here exactly (same for the revertedLinksMatch)?